### DISCUSSION TITLE
Vampiric Touch Overhaul

### DISCUSSION THREAD

**Discussion by tenaz123** [DarkCastleMUD/DarkCastle#238] [General]
Why is vampiric touch gimped so much?  There are a lot of creatures who are immune/resistant to cold (the same are also immune/resistant to acid for some odd reason), and on top of this, vamp touch is further hampered by something (not saves) reducing the damaged randomly to 1/4, 1/2 or 3/4 even on the lowest of mobs.  See the copy paste below where the first hits for normal damaged, the 2nd for half without a save, and the 3rd a regular resisted one with a 10% reduction in damage.  When there's a resist cold save, you get the normal % reduction like other spells, but on top of this, you'll get the spell doing the lower damaged without a saves message.

I can only imagine this was done due to weapons with vamp touch on them, but I wonder why weapons have this spell anyway, at least weapons that are unaligned.  This spell is only available to evil anti-pals, yet the thief quest weapon gives this heal ability to what is already the best class.  At best, vamp touch should only appear on evil only weapons and definitely not on a thief only weapon.  This gimping also hampers Vamp Aura.

Acid blast is also an evil only anti spell that shows up on lots of weapons.  I'm not sure how much I'd consider that innately evil though since fire/lightning etc can all disfigure.  It seems like a spell mages should have in their arsenal.

Ok.
A White Rook crumples as you touch him and drain the remaining 219 of his lifeforce.
A White Rook is DEAD!!
You receive 4850 exps of 4850 total.
Your blood freezes as you hear a White Rook's death cry.

Ok.
A White Knight crumples as you touch him and drain the remaining 119 of his lifeforce.
A White Knight is DEAD!!
You receive 6435 exps of 6435 total.
Your blood freezes as you hear a White Knight's death cry.

Erawan resists your cold and sustains reduced damage.
You drain Erawan of 206 lifeforce by touching him.

**Comment by TehDreadPirateRoberts** [comment]
Vampiric touch has something interesting in it. It has a check for saves right within the code for it that has a chance then to reduce the damage by 1/2. This is separate code from the code within fight.cpp where it checks for if someone is resistant and you get the "...resists your cold and sustains reduced damage." message.
I think this might be worth testing removing lines 514 to 518 in magic.cpp and testing a bunch of casts of vampiric touch.

**Reply by tenaz123** [reply]
You really do want to give me a reason to spin up a test server haha.
